src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -451,7 +455,7 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     for(unsigned j=1; j<size; j++)
     {
       mult_exprt radix_sum(sum, radix);
-      if(j>=9)
+      if(j>=max_string_length-1)
```

Comment:
As per our discussion, I will leave this detail because we are rewriting this function in https://github.com/diffblue/test-gen/issues/702